crates/prover/src/runner.rs

```diff
@@ -223,6 +223,29 @@ where
         });
     }
 
+    async fn check_and_recover_ongoing_proving_sessions(
+        &self,
+        l1_height: u64,
+    ) -> Result<bool, anyhow::Error> {
+        let prover_service = self
+            .prover_service
+            .as_ref()
+            .expect("Prover service should be present");
+        let results = prover_service
+            .recover_proving_sessions_and_send_to_da(&self.da_service)
+            .await?;
+        if results.is_empty() {
+            Ok(false)
+        } else {
+            for (tx_id, proof) in results {
+                self.extract_and_store_proof(tx_id, proof, l1_height)
+                    .await?;
+            }
+            self.ledger_db.clear_pending_proving_sessions()?;
```

Comment:
we might want to do this inside prover service if possible.

Reply:
done